tests/testthat/helper-setup.R

```diff
@@ -0,0 +1,235 @@
+
+# Code that needs to run once, before a suite of tests is run.
+# Here, "suite of tests" might also mean "a single test" interactively.
+renv_tests_setup <- function(envir = parent.frame()) {
+
+  # ensure that attempts to restart are a no-op
+  if (renv_rstudio_available())
+    options(restart = function(...) NULL)
+
+  # cache path before working directory gets changed
+  renv_tests_root()
+
+  # make sure required packages are loaded
+  # (not scoped to the environment since packages can't reliably be unloaded)
+  renv_tests_setup_packages()
+
+  # initialize test repositories
+  renv_tests_setup_repos(envir = envir)
+
+  # scope relevant environment variables
+  renv_tests_setup_envvars(envir = envir)
+  renv_tests_setup_options(envir = envir)
+
+  # set up interactive tools
+  renv_tests_setup_tools(envir)
+
+}
+
+
+renv_tests_setup_envvars <- function(envir = parent.frame()) {
+
+  # set up root directory
+  root <- normalizePath(
+    ensure_directory(renv_scope_tempfile(envir = envir)),
+    mustWork = FALSE,
+    winslash = "/"
+  )
+
+  renv_scope_envvars(
+    # simulate running in R CMD check
+    "_R_CHECK_PACKAGE_NAME_" = "renv",
+    # disable locking in this scope
+    RENV_CONFIG_LOCKING_ENABLED = FALSE,
+    RENV_PATHS_ROOT = root,
+    RENV_PATHS_LIBRARY = NULL,
+    RENV_PATHS_LIBRARY_ROOT = NULL,
+    RENV_PATHS_LOCAL = NULL,
+    RENV_PATHS_LOCKFILE = NULL,
+    RENV_PATHS_RENV = NULL,
+    RENV_AUTOLOAD_ENABLED = FALSE,
+    envir = envir
+  )
+
+  if (is.na(Sys.getenv("GITHUB_PATH", unset = NA))) {
+    token <- tryCatch(gitcreds::gitcreds_get(), error = function(e) NULL)
+    if (!is.null(token)) {
+      renv_scope_envvars(GITHUB_PAT = token$password, envir = envir)
+    }
+  }
+
+  envvars <- Sys.getenv()
+  configvars <- grep("^RENV_CONFIG_", names(envvars), value = TRUE)
+  renv_scope_envvars(
+    list = rep_named(configvars, list(NULL)),
+    envir = envir
+  )
+}
+
+renv_tests_setup_options <- function(envir = parent.frame()) {
+
+  renv_scope_options(
+    renv.bootstrap.quiet = TRUE,
+    renv.config.user.library = FALSE,
+    renv.config.sandbox.enabled = TRUE,
+    restart = NULL,
+    renv.config.install.transactional = FALSE,
+    renv.tests.running = TRUE,
+    envir = envir
+  )
+
+}
+
+renv_tests_setup_tools <- function(envir) {
+
+  if (interactive()) {
+
+    # create a 'done' object that, when printed, will
+    # run any pending defer handlers
+    envir <- attach(NULL, name = "renv:tools")
```

Comment:
Given that you're changing from an active binding to a print method, are you sure that it needs to go into a special environment?

But TBH it feels like that this is a lot of gymnastics to save typing two characters, which autocomplete would provide for you anyway.